lib/ex_doc/autolink.ex

```diff
@@ -92,17 +89,37 @@ defmodule ExDoc.Autolink do
   end
 
   defp custom_link(attrs, config) do
-    with {:ok, href} <- Keyword.fetch(attrs, :href),
-         [[_, text]] <- Regex.scan(~r/^`(.+)`$/, href) do
-      url(text, :custom_link, config)
-    else
-      _ -> nil
+    case Keyword.fetch(attrs, :href) do
+      {:ok, href} ->
+        case Regex.scan(~r/^`(.+)`$/, href) do
+          [[_, custom_link]] ->
+            build_custom_link(custom_link, config)
+
+          [] ->
+            build_extra_link(href, config)
+        end
+
+      _ ->
+        nil
+    end
+  end
+
+  defp build_custom_link(link, config) do
+    cond do
+      url = url(link, :custom_link, config) ->
+        url
+
+      true ->
+        uri = URI.parse(link)
```

Comment:
what's the reason for URI.parse/1 here? By this point, we are supposed to be between `` `(...)` `` here anyway. If it's needed, could you add a test?